src/Chainweb/CutDB.hs

```diff
@@ -297,31 +303,41 @@ processCuts logFun headerStore payloadStore queue cutVar = queueToStream
         )
     & S.effects
   where
+    -- | Broadcast the newest estimated `BlockHeight` of the network to other
+    -- components of this node.
+    --
+    updateNetworkHeight :: CutHashes -> IO ()
+    updateNetworkHeight = atomically . writeTVar networkHeight . Just . cutHashesHeight
+
     graph = _chainGraph headerStore
 
     threshold :: Int
     threshold = int $ 2 * diameter graph * order graph
 
+    queueToStream :: S.Stream (Of CutHashes) IO ()
     queueToStream = do
         Down a <- liftIO (pQueueRemove queue)
         S.yield a
         queueToStream
 
+    isVeryOld :: CutHashes -> IO Bool
     isVeryOld x = do
         h <- _cutHeight <$> readTVarIO cutVar
-        let r = int (_cutHashesHeight x) <= (int h - threshold)
+        let !r = int (_cutHashesHeight x) <= (int h - threshold)
```

Comment:
I don't think that the BangPattern is needed here, since we discriminate on the value in next line. But it doesn't hurt neither.